pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -132,8 +143,13 @@ func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context,
 		return fmt.Errorf("could not list etcd members: %w", err)
 	}
 
+	// removing etcd-boostrap from live voting members count for HAScalingStrategy
+	if scalingStrategy == ceohelpers.HAScalingStrategy {
+		liveVotingMembers = removeBootstrapMember(liveVotingMembers)
+	}
+
 	if len(liveVotingMembers) < desiredControlPlaneReplicasCount {
-		klog.V(2).Infof("Ignoring scale-down since the number of etcd members (%d) < desired number of control-plane replicas (%d) ", len(liveVotingMembers), desiredControlPlaneReplicasCount)
+		klog.V(2).Infof("Ignoring scale-down since the number of voting, non-bootstrap etcd members is (%d) < desired number of control-plane replicas (%d)", len(liveVotingMembers), desiredControlPlaneReplicasCount)
```

Comment:
The "non-bootstrap" qualifier [I'd asked for earlier][1] is a bit fiddly now that `removeBootstrapMember` is conditioned on `scalingStrategy == ceohelpers.HAScalingStrategy`.  Maybe use something more generic like "the number of production etcd members"?  Or use a string variable to hold the relevant specific qualifier:

```go
memberQualifier := "voting"
if scalingStrategy == ceohelpers.HAScalingStrategy {
	memberQualifier = "voting, non-bootstrap"
	liveVotingMembers = removeBootstrapMember(liveVotingMembers)
}
```	

and then add that to logs with `... number of %s etcd members...`?

[1]: https://github.com/openshift/cluster-etcd-operator/pull/1027#discussion_r1139351212

Reply:
thanks @wking for the review

May I ask what is the benefit from `memberQualifier = "voting, non-bootstrap` ? 

I check the scaling strategy in the beginning of the control-loop (i.e. sync), and use this assertion before deducting the `etcd-bootstrap` member from the voting-members. 

iiuc, we need not deducing the boot-strap count from the membership in other strategies, please correct me if I am wrong :)

Reply:
I am not sure why this new logic is not relevant for delayed strategy but message is not right in that case as you will have bootstrap member in case it is not HaScalingStrategy, right?

Reply:
But why this not needed for DelayedHAScalingStrategy?

Reply:
Thanks @tsorya for your review 

iiuc, we trying to maintain the etcd membership for HAScalingStrategy preserving the quorum. CPMS triggered master machine replacement while the boot-strap member still exist in the etcd cluster membership. This caused [bug](https://issues.redhat.com/browse/OCPBUGS-8038).

for DelayedHAScalingStrategy, I am not sure if similar issue was found 


cc @JoelSpeed @damdo

Reply:
I also don't like that `if scalingStrategy == ceohelpers.HAScalingStrategy {` littered all over the place, it's very hard to follow the logic. Can you please add this to `CheckSafeToScaleCluster`?

Reply:
> I also don't like that `if scalingStrategy == ceohelpers.HAScalingStrategy {` littered all over the place, it's very hard to follow the logic. Can you please add this to `CheckSafeToScaleCluster`?

so i skip the sync loop for other strategies 🤔  .. cant get what you mean here

Reply:
> so i skip the sync loop for other strategies  .. cant get what you mean here

we just settled in-person that we rely on the BootstrapTearDownController to determine the safety of the scale-down. As long as there's a bootstrap member it's never safe to scale down. We can implicitly rely on this from this controller.

That will safe us all the math and extra checks for strategy as this is already in the other controller.

Reply:
it broke the SNO :/ 

Fixing ....